src/glfw3.jl

```diff
@@ -289,12 +290,31 @@ immutable VidMode
 	refreshrate::Cint   # The refresh rate, in Hz, of the video mode.
 end
 
+immutable GLFWError <: Exception
+	code::Cint
+	description::String
+end
+Base.showerror(io::IO, e::GLFWError) = print(io, "GLFWError: ", e.description)
+
 #************************************************************************
 # GLFW API functions
 #************************************************************************
 
 # Initialization and version information
-Init() = Bool(ccall( (:glfwInit, lib), Cint, ())) || error("initialization failed")
+function Init()
+	try
+		return Bool(ccall( (:glfwInit, lib), Cint, ())) || error("initialization failed")
+	catch ex
+		if isa(ex, GLFWError) && ex.code == PLATFORM_ERROR && contains(ex.description, "Failed to get display service port iterator")
```

Comment:
This won't be enough, since there is also the RandR issue:
https://github.com/glfw/glfw/blob/master/src/x11_init.c#L537